types/config_test.go

```diff
@@ -68,3 +71,53 @@ func (s *configTestSuite) TestConfig_SetFullFundraiserPath() {
 func (s *configTestSuite) TestKeyringServiceName() {
 	s.Require().Equal(sdk.DefaultKeyringServiceName, sdk.KeyringServiceName())
 }
+
+func (s *configTestSuite) TestConfig_ScopePerBinary_DefaultBehavior() {
+	cfg1 := sdk.GetConfig()
+	cfg2 := sdk.GetConfig()
+	s.Require().Equal(cfg1, cfg2, "configs should be identical in same binary by default")
+}
+
+func (s *configTestSuite) TestConfig_ScopePerBinary_EnvOverride() {
+	original := os.Getenv(sdk.EnvConfigScope)
+	defer os.Setenv(sdk.EnvConfigScope, original)
+
+	s.Require().NoError(os.Setenv(sdk.EnvConfigScope, "test-scope-A"))
+	cfgA := sdk.GetConfig()
+
+	s.Require().NoError(os.Setenv(sdk.EnvConfigScope, "test-scope-B"))
+	cfgB := sdk.GetConfig()
+
+	s.Require().NotEqual(cfgA, cfgB, "configs should differ for different env scopes")
+}
+
+func (s *configTestSuite) TestConfig_ScopePerBinary_EnvRestoration() {
+	envKey := sdk.EnvConfigScope
+	original := os.Getenv(envKey)
+	defer os.Setenv(envKey, original)
+
+	s.Require().NoError(os.Setenv(envKey, "test-scope-Restore"))
+	cfg1 := sdk.GetConfig()
+
+	s.Require().NoError(os.Setenv(envKey, "test-scope-Restore"))
+	cfg2 := sdk.GetConfig()
+
+	s.Require().Equal(cfg1, cfg2, "config should remain stable with same env scope")
+}
+
+func (s *configTestSuite) TestConfig_ScopeKeyFormat() {
+	key := getTestConfigKey()
+	s.Require().True(strings.Count(key, "|") == 2, "scope key should have 2 pipe separators")
+}
+
+// getTestConfigKey replicates the internal key generation logic for testing visibility.
+func getTestConfigKey() string {
+	id := os.Getenv(sdk.EnvConfigScope)
+	if id != "" {
+		return id
+	}
+	exe, _ := os.Executable()
+	host, _ := os.Hostname()
+	pid := os.Getpid()
+	return fmt.Sprintf("%s|%s|%d", host, exe, pid)
+}
```

Comment:
we can avoid this by testing in `package types` rather than `package types_test`.

Reply:
updated